### DISCUSSION TITLE
How to distinguish TE and TM in a multilayer film?

### DISCUSSION THREAD

**Discussion by niubiworker** [NanoComp/meep#2915] [Q&A]
Most photonic crystal slabs are symmetric along the z-direction (with the xy-plane as the periodic plane), but now I want to design a multilayer structure (such as air, silicon, silicon dioxide, air), where the periodicity remains in the xy-plane or along a line-defect waveguide in a certain direction. Clearly, the symmetry in the z-direction is broken in this multilayer structure. Can I still use the mpb function TE-yodd to distinguish between TE or TM modes?Thank you very much!

**Comment by stevengj** [comment]
No.   In this case there is no z mirror plane, and hence no strict symmetry classification in z.

Even when there *is* a strict z=0 mirror symmetry plane, the even/odd modes are not *purely* TE/TM polarized.  See also this figure from [our book](http://ab-initio.mit.edu/book/):
![image](https://github.com/user-attachments/assets/6a74f0ac-f0fc-4600-b6ef-a002faf1b7cf)

However, if you have another mirror symmetry plane, you can use that.  For example, for a line defect along the x direction, you may have a y=0 mirror-symmetry plane, and you can use that to help you classify the modes.
